ee/modules/610-service-with-healthchecks/crds/servicewithhealthchecks.yaml

```diff
@@ -0,0 +1,785 @@
+
+                  The Kubernetes controllers that generate `Endpoints` and `EndpointSlice` resources for
+                  Services interpret this to mean that all endpoints are considered "ready" even if the
+                  Pods themselves are not.
+                  Agents which consume only Kubernetes generated endpoints
+                  through the `Endpoints` or `EndpointSlice` resources can safely assume this behavior.
+                type: boolean
+              selector:
+                additionalProperties:
+                  type: string
+                description: |-
+                  Route service traffic to pods with label keys and values matching this
+                  selector. If empty or not present, the service is assumed to have an
+                  external process managing its endpoints, which Kubernetes will not
+                  modify.
+
+                  Only applies to types ClusterIP, NodePort, and LoadBalancer.
+
+                  Ignored if type is `ExternalName`.
+                  [More info.](https://kubernetes.io/docs/concepts/services-networking/service/)
+                type: object
+                x-kubernetes-map-type: atomic
+              sessionAffinity:
+                description: |-
+                  Used to maintain session affinity.
+                  Enable client IP based session affinity.
+                  Must be `ClientIP` or `None`.
+                  [More info.](https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies)
+                type: string
+              sessionAffinityConfig:
+                description: Contains the configurations of session affinity.
+                properties:
+                  clientIP:
+                    description: Contains the configurations of `ClientIP` based session affinity.
+                    properties:
+                      timeoutSeconds:
+                        description: |-
+                          Specifies the seconds of `ClientIP` type session sticky time.
+                          The value must be `>0 && <=86400` (for 1 day) if `ServiceAffinity == ClientIP`.
+                        format: int32
+                        type: integer
+                    type: object
+                type: object
+              trafficDistribution:
+                description: |-
+                  Offers a way to express preferences for how traffic is
+                  distributed to Service endpoints. Implementations can use this field as a
+                  hint, but are not required to guarantee strict adherence.
+
+                  If the field is
+                  not set, the implementation will apply its default routing strategy.
+                  If set
+                  to `PreferClose`, implementations should prioritize endpoints that are
+                  topologically close (e.g., same zone).
+                type: string
+              type:
```

Comment:
Enum?

Reply:
We do our best to preserve the identity between the specifications of the two resources